packages/mui-material/src/TextField/TextField.js

```diff
@@ -142,12 +142,30 @@ const TextField = React.forwardRef(function TextField(inProps, ref) {
   const externalForwardedProps = {
     slots,
     slotProps: {
-      input: InputPropsProp,
-      inputLabel: InputLabelPropsProp,
-      htmlInput: inputPropsProp,
-      formHelperText: FormHelperTextPropsProp,
-      select: SelectPropsProp,
       ...slotProps,
```

Comment:
Issue is props which are set by `Autocomplete` to `input` slot is getting overriden  by `slotProps` object. Instead of overrding i tried to merge `InputProps` and `slotProps.htmlInput` and looks like it is working

https://github.com/mui/material-ui/blob/f8b2251c5bdfe4d81f56017e0423d265b242617d/packages/mui-material/src/Autocomplete/Autocomplete.js#L736